core/controllers/topic_editor_test.py

```diff
@@ -552,6 +552,30 @@ def test_assign_topic_manager_role(self):
                 {}, csrf_token=csrf_token, expected_status_int=401)
 
 
+class TopicPublishSendMailHandlerTests(BaseTopicEditorControllerTests):
+
+    def test_send_mail(self):
+        self.login(self.ADMIN_EMAIL)
+        response = self.get_html_response(
+            '%s/%s' % (feconf.TOPIC_EDITOR_URL_PREFIX, self.topic_id))
+        csrf_token = self.get_csrf_token_from_response(response)
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            self.put_json(
+                '%s/%s' % (
+                    feconf.TOPIC_SEND_MAIL_URL_PREFIX, self.topic_id),
+                {'topic_name': 'Topic Name'}, csrf_token=csrf_token)
+            messages = self.mail_stub.get_sent_messages(
```

Comment:
I think you can pull this and everything else below it out of the "with" block. Generally keep the "with" block as minimal as possible.

Reply:
Done.